pkg/connections/connections.go

```diff
@@ -0,0 +1,331 @@
+package connections
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path"
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/openshift-kni/commatrix/pkg/client"
+	"github.com/openshift-kni/commatrix/pkg/consts"
+	"github.com/openshift-kni/commatrix/pkg/types"
+	"github.com/openshift-kni/commatrix/pkg/utils"
+)
+
+const (
+	localAddrPortFieldIdx = 3
+	interval              = time.Millisecond * 500
+	duration              = time.Second * 5
+)
+
+type ConnectionCheck struct {
+	*client.ClientSet
+	podUtils            utils.UtilsInterface
+	customEntriesPath   string
+	customEntriesFormat string
+	format              string
+	destDir             string
+	env                 types.Env
+	deployment          types.Deployment
+	nodesRolesMap       map[string][]string
+}
+
+func NewConnectionCheck(c *client.ClientSet, podUtils utils.UtilsInterface, customEntriesPath, customEntriesFormat, format, destDir string, env types.Env, deployment types.Deployment) (*ConnectionCheck, error) {
+	nodeList := &corev1.NodeList{}
+	err := c.List(context.TODO(), nodeList)
+	if err != nil {
+		return nil, err
+	}
+
+	nodesRolesMap := map[string][]string{}
+	for _, node := range nodeList.Items {
+		nodesRolesMap[node.Name] = types.GetNodeRoles(&node)
+	}
+
+	return &ConnectionCheck{
+		c,
+		podUtils,
+		customEntriesPath,
+		customEntriesFormat,
+		format,
+		destDir,
+		env,
+		deployment,
+		nodesRolesMap,
+	}, nil
+}
+
+func (cc *ConnectionCheck) GenerateSS() (*types.ComMatrix, []byte, []byte, error) {
+	nodesList := &corev1.NodeList{}
```

Comment:
why not to save the node names on the ConnectionCheck? 
as the node role list u can also save the node name and we will not use the corv1 to list

Reply:
I like the idea nice!